test/runtests.jl

```diff
@@ -1,88 +1,163 @@
     @test !UpdateJulia.prefer(v"1.7.0-rc1", v"1.6.4")
     @test !UpdateJulia.prefer(v"17.7.0-rc1", v"1.6.4")
     @test UpdateJulia.prefer(v"17.7.0-rc1", v"1.6.4-DEV")
     @test UpdateJulia.prefer(v"17.7.0-rc1", v"2.6.4-DEV")
+    @test !UpdateJulia.prefer(v"17.7.0-DEV", v"2.6.4-rc1")
 
     @test !UpdateJulia.prefer(missing, v"1.6.4")
     @test !UpdateJulia.prefer(missing, missing)
     @test UpdateJulia.prefer(v"1.7.0-rc1", missing)
 end
 
-@testset "UpdateJulia.jl" begin
+@testset "latest()" begin
+    @test UpdateJulia.latest() >= v"1.7.0"
+    @test UpdateJulia.latest().prerelease == ()
 
-    display(@doc update_julia)
-    println()
-    #TODO optimize testing
+    l = UpdateJulia.latest()
+    for v in keys(UpdateJulia.versions[])
+        @test v === l || UpdateJulia.prefer(l, v)
+    end
+end
 
-    printstyled("WARNING: this will both install out of date julia versions and change what the command `julia` points to.\n", color=Base.warn_color())
-    printstyled("If all goes well, it will finish with the latest stable version of julia installed.\n", color=Base.warn_color())
+@testset "versions[]" begin
+    @test VersionNumber[v"0.1.2", v"0.3.1", v"0.5.0-rc2", v"0.6.0-pre.beta", v"0.7.0-beta",
+        v"1.0.0", v"1.1.1", v"1.2.0-rc2", v"1.2.0-rc3", v"1.2.0", v"1.3.0-rc3", v"1.4.1",
+        v"1.4.2", v"1.5.1", v"1.5.4", v"1.6.0-rc3", v"1.6.1", v"1.6.4", v"1.7.0-beta2",
+        v"1.7.0-beta3"] ⊊ keys(UpdateJulia.versions[])
+end
 
-    global v_latest = update_julia()
-    @test v_latest == UpdateJulia.latest()
-    @test v_latest >= v"1.6.4" && v_latest.prerelease == ()
-
-    @static if Sys.iswindows()
-        @test_skip "interactive windows installer"
-    else
-        @test update_julia("1.4") == v"1.4.2" # TODO check that this reports succeeds on all three counts
-        @test update_julia("1.2") == v"1.2.0"
-        @test update_julia("1.0.4") == v"1.0.4"
-        global v10_latest = update_julia("1.0")
-        @test v10_latest >= v"1.0.5" && v10_latest < v"1.1.0"
-        @test update_julia("1.0.0") == v"1.0.0"
-    end
+@testset "dry tests" begin
+    # Basics
+    @test update_julia(dry_run = true) == UpdateJulia.latest()
+    @suppress_out @test UpdateJulia.update_julia("nightly", dry_run=true, verbose=false) ==
+        UpdateJulia.update_julia(string(UpdateJulia.nightly_version[]), dry_run=true, verbose=false) ==
+        UpdateJulia.nightly_version[]
+
+    # Speed
+    f() = @test @suppress update_julia("1.7.0-rc1", dry_run=true, verbose=false) == v"1.7.0-rc1"
+    @elapsed f()
+    @test .05 > @elapsed f()
 
-    Sys.iswindows() || println("requesting GUI")
-    @test update_julia("1.5.2", prefer_gui = !Sys.iswindows()) == v"1.5.2" # not supported. test fallback.
-    @test update_julia("1.5.3", systemwide=true, verbose=true) == v"1.5.3" # TODO check that this reports succeeds on all three counts
-    @test update_julia("1.5.1", systemwide=false) == v"1.5.1"
-    @test update_julia() == v_latest
```

Comment:
In the matrix... add to curated with prefer_gui.

Reply:
We have them in both orders now: https://github.com/LilithHafner/UpdateJulia.jl/blob/50c993c3ad80543c32d723e419486e08402f5023/test/runtests.jl#L215